src/go/k8s/apis/redpanda/v1alpha1/console_types.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ConsoleSpec defines the desired state of Console
+type ConsoleSpec struct {
+	// +optional
+	// +kubebuilder:default=console
+	MetricsNamespace string `json:"metricsNamespace"`
+
+	// +optional
+	// +kubebuilder:default=true
+	ServeFrontend bool `json:"serveFrontend"`
+
+	// +optional
+	Server Server `json:"server"`
+
+	Schema        Schema                 `json:"schema"`
+	ClusterKeyRef corev1.ObjectReference `json:"clusterKeyRef"`
+	Deployment    Deployment             `json:"deployment"`
+	Connect       Connect                `json:"connect"`
+}
+
+// Server is the Console app HTTP server config
+type Server struct {
+	// +kubebuilder:default="30s"
+	ServerGracefulShutdownTimeout string `json:"gracefulShutdownTimeout,omitempty" yaml:"gracefulShutdownTimeout,omitempty"`
+
+	HTTPListenAddress string `json:"listenAddress,omitempty" yaml:"listenAddress,omitempty"`
+
+	// +kubebuilder:default=8080
+	HTTPListenPort int `json:"listenPort,omitempty" yaml:"listenPort,omitempty"`
+
+	// +kubebuilder:default="30s"
+	HTTPServerReadTimeout string `json:"readTimeout,omitempty" yaml:"readTimeout,omitempty"`
+
+	// +kubebuilder:default="30s"
+	HTTPServerWriteTimeout string `json:"writeTimeout,omitempty" yaml:"writeTimeout,omitempty"`
+
+	// +kubebuilder:default="30s"
+	HTTPServerIdleTimeout string `json:"idleTimeout,omitempty" yaml:"idleTimeout,omitempty"`
+
+	// +kubebuilder:default=4
+	CompressionLevel int `json:"compressionLevel,omitempty" yaml:"compressionLevel,omitempty"`
+
+	BasePath string `json:"basePath,omitempty" yaml:"basePath,omitempty"`
+
+	// +kubebuilder:default=true
+	SetBasePathFromXForwardedPrefix bool `json:"setBasePathFromXForwardedPrefix,omitempty" yaml:"setBasePathFromXForwardedPrefix,omitempty"`
+
+	// +kubebuilder:default=true
+	StripPrefix bool `json:"stripPrefix,omitempty" yaml:"stripPrefix,omitempty"`
+}
+
+// Schema defines configurable fields for Schema Registry
+type Schema struct {
```

Comment:
nit: can we name this SchemaRegistry

Reply:
Marking as resolved [REF](https://github.com/redpanda-data/redpanda/pull/5758/commits/12e2650d68bee7b33ad5c170dc59580825584021)